tsnake/grid.py

```diff
@@ -322,22 +322,39 @@ def get_snake_intersections(self, snakes: TSnakes) -> [[Point]]:
         """
         return [self._compute_intersection(snake) for snake in snakes]
 
-    def reparameterize(self, snakes: TSnakes):
+    def reparameterize_phase_one(self, snakes: TSnakes):
         '''
-        Takes a list of T-Snake instances and returns a new list of TSnakes
-        (which have possibly been split/merged/etc).
-        If no snakes are split/merged, then:
-        1) the snakes are presented in the same order as before, AND
-        2) each snake's nodes are presented in the same order.
+        Takes a list of T-Snakes and returns a new list of T-Snakes (which have possibly been split/merged/etc).
```

Comment:
This isn't an outline, it's just re-typing more or less the exact text of the paper. All for it if it helps you understand what to do, but it's really time for the rubber to hit the road here, I've actually written the functional skeleton for this in the meantime on my end.